eth-supply/src/lib.rs

```diff
@@ -0,0 +1,103 @@
+#[rustfmt::skip]
+pub mod abi;
+#[rustfmt::skip]
+pub mod pb;
+#[rustfmt::skip]
+pub mod helpers;
+
+use std::ops::Add;
+use std::ops::Sub;
+use std::str::FromStr;
+
+use pb::eth_supply::v1 as eth_supply;
+use pb::eth_supply::v1::EthSupply;
+
+use substreams::scalar::BigInt;
+use substreams::store;
+use substreams::store::StoreAdd;
+use substreams::store::StoreAddBigInt;
+use substreams::store::StoreGet;
+use substreams::store::StoreGetBigInt;
+use substreams::store::StoreNew;
+use substreams_ethereum::pb::eth as pbeth;
+
+use helpers::{get_delta, get_eip1559_burn, get_genesis_value};
+
+#[substreams::handlers::map]
+fn map_supply_delta(
+    block: pbeth::v2::Block,
+) -> Result<eth_supply::EthSupply, substreams::errors::Error> {
+    let mut block_rewards: BigInt = BigInt::from(0);
+    let mut uncle_rewards: BigInt = BigInt::from(0);
+    let genesis: BigInt = get_genesis_value(&block);
+
+    for change in block.balance_changes {
+        match pbeth::v2::balance_change::Reason::from_i32(change.reason).unwrap_or_default() {
+            pbeth::v2::balance_change::Reason::RewardMineUncle => {
+                uncle_rewards = uncle_rewards.add(get_delta(change));
+            }
+            pbeth::v2::balance_change::Reason::RewardMineBlock => {
+                block_rewards = block_rewards.add(get_delta(change));
+            }
+            _ => {}
+        }
+    }
+
+    let burned = get_eip1559_burn(&block.header.unwrap());
```

Comment:
For stuff like this use block.header.as_ref().unwrap() instead

Reply:
Nothing against it, but curious about the why :P

Reply:
So what you had before is fine and would work however it's best to use this way as a good habbit. The reason is that if you don't use .as_ref() before you unwrap an option you take ownership over the option. So your way essentially takes ownership of the field and then takes a reference again from the moved value. Under the hood I'm sure the compiler would optimise out the difference in implementation however in other scenarios this could give you various errors with the borrow checker when it comes to the taking of ownership here. So yes this works although i'd say as_ref() would be better practise here